go-controller/pkg/ovn/policy.go

```diff
@@ -17,17 +17,19 @@ import (
 
 type namespacePolicy struct {
 	sync.Mutex
-	name            string
-	namespace       string
-	policyTypes     []knet.PolicyType
-	ingressPolicies []*gressPolicy
-	egressPolicies  []*gressPolicy
-	podHandlerList  []*factory.Handler
-	nsHandlerList   []*factory.Handler
-	localPods       map[string]*lpInfo //pods effected by this policy
-	portGroupUUID   string             //uuid for OVN port_group
-	portGroupName   string
-	deleted         bool //deleted policy
+	name                 string
+	namespace            string
+	policyTypes          []knet.PolicyType
+	portGroupIngressDeny string // Port group for ingress deny rule
+	portGroupEgressDeny  string // Port group for egress deny rule
```

Comment:
This ends up switching from having a single cluster-wide deny group to having a deny group per namespace, even if logging isn't being used.

If we don't keep per-namespace log level configuration then you just need to have one port group for namespaces that log, and one for namespaces that don't. If you do keep per-namespace log levels then you could still just have one deny group per logging level (eg, a "deny and log at level `info`" group, etc).

I guess that would mean that when enabling/disabling logging on a namespace you'd have to move all the pods from one port group into another? Maybe that would be bad? Not sure what the best tradeoff here is...

Reply:
yes, I wanted to avoid moving things and thought that impact of having per-namespace PGs is minimal if at all. I've noticed that the way OVSDB handles "find" / "update" requests today that are not _uuid  based, are O(N), which makes things unnecessary slow if we use large tables. My thought here is that by introducing per-namespace deny PGs we could improve OVSDB requests and keep plumbing logic simple, i.e. PGs per name-space even if logging is disabled.

Reply:
What is the upper bound for OVSDB query's that are `_uuid` based? I can't think of any obvious arguments against per namespace deny port groups, except for the slightly smaller overhead. It will also be easier to modulate the code moving forward... i.e pushing `policy.go` to it's own package